src/Model/ModelManager.php

```diff
@@ -458,7 +472,7 @@ public function modelReverseTransform($class, array $array = [])
                 $instance->$property = $value;
             } elseif ($reflClass->hasProperty($property)) {
                 if (!$reflClass->getProperty($property)->isPublic()) {
-                    throw new PropertyAccessDeniedException(sprintf('Property "%s" is not public in class "%s". Maybe you should create the method "set%s()"?', $property, $reflClass->getName(), ucfirst($property)));
+                    throw new \BadMethodCallException(sprintf('Property "%s" is not public in class "%s". Maybe you should create the method "set%s()"?', $property, $reflClass->getName(), ucfirst($property)));
```

Comment:
Since the checked element here is a property, I think `\BadMethodCallException` is not the proper exception.

Reply:
Now that you point this out, rethinking about this, maybe we can just use `PropertyAccessor` class instead of all this code.

```diff
-        $reflClass = $metadata->reflClass;
+        $propertyAccesor = PropertyAccess::createPropertyAccessorBuilder()
+            ->enableMagicCall()
+            ->getPropertyAccessor();
+
         foreach ($array as $name => $value) {
-            $reflection_property = false;
             // property or association ?
             if (\array_key_exists($name, $metadata->fieldMappings)) {
                 $property = $metadata->fieldMappings[$name]['fieldName'];
-                $reflection_property = $metadata->reflFields[$name];
             } elseif (\array_key_exists($name, $metadata->associationMappings)) {
                 $property = $metadata->associationMappings[$name]['fieldName'];
             } else {
                 $property = $name;
             }

-            $setter = 'set'.$this->camelize($name);
-
-            if ($reflClass->hasMethod($setter)) {
-                if (!$reflClass->getMethod($setter)->isPublic()) {
-                    throw new \BadMethodCallException(sprintf('Method "%s()" is not public in class "%s"', $setter, $reflClass->getName()));
-                }
-
-                $instance->$setter($value);
-            } elseif ($reflClass->hasMethod('__set')) {
-                // needed to support magic method __set
-                $instance->$property = $value;
-            } elseif ($reflClass->hasProperty($property)) {
-                if (!$reflClass->getProperty($property)->isPublic()) {
-                    throw new \BadMethodCallException(sprintf('Property "%s" is not public in class "%s". Maybe you should create the method "set%s()"?', $property, $reflClass->getName(), ucfirst($property)));
-                }
-
-                $instance->$property = $value;
-            } elseif ($reflection_property) {
-                $reflection_property->setValue($instance, $value);
-            }
+            $propertyAccesor->setValue($instance, $property, $value);
         }
```
The problem I see is that I don't know how to reach this case:
```php
$reflection_property->setValue($instance, $value);
```
Or change the exception, any suggestion? I've looking for one, but haven't found one that I like.

Reply:
So I think we could replace this with the `PropertyAccess` class, because what I understand is... 

it checks first if there is a public setter:

```php
$setter = 'set'.$this->camelize($name);

if ($reflClass->hasMethod($setter)) {
    if (!$reflClass->getMethod($setter)->isPublic()) {
        throw new \BadMethodCallException(sprintf('Method "%s()" is not public in class "%s"', $setter, $reflClass->getName()));
    }

    $instance->$setter($value);
```
then checks the `__set` method:
```php
} elseif ($reflClass->hasMethod('__set')) {
    // needed to support magic method __set
    $instance->$property = $value;
```
and then a public property:
```php
} elseif ($reflClass->hasProperty($property)) {
    if (!$reflClass->getProperty($property)->isPublic()) {
        throw new \BadMethodCallException(sprintf('Property "%s" is not public in class "%s". Maybe you should create the method "set%s()"?', $property, $reflClass->getName(), ucfirst($property)));
    }

    $instance->$property = $value;
```
and finally this:
```php
} elseif ($reflection_property) {
    $reflection_property->setValue($instance, $value);
}
```
I think that `$reflection_property->setValue($instance, $value);` is covered by the previous cases and it would fail if the property is not accesible. So basically I think it could be all replaced by [`PropertyAccess`](https://symfony.com/doc/4.4/components/property_access.html#writing-to-objects) and the exceptions wouldn't be a problem since the classes don't exist anymore.